erazhu31/TLS Notary DOracle/src/oracle.js

```diff
@@ -0,0 +1,43 @@
+const https   = require('https');
+const ethers  = require('ethers');
+const fs      = require('fs');
+
+const sslCertificate = require('get-ssl-certificate')
+
+
+const root                = 'iexec_out';
+const determinismFilePath = `${root}/determinism.iexec`;
+const callbackFilePath    = `${root}/callback.iexec`;
+const errorFilePath       = `${root}/error.iexec`;
+
+
+
+
+
+/*****************************************************************************
+ *                                 ARGUMENTS                                 *
+ *****************************************************************************/
+let URL = process.argv.slice(2);
+
+
+
+
+/*****************************************************************************
+ *                                  EXECUTE                                  *
+ *****************************************************************************/
+
+sslCertificate.get(`${URL}`).then(function (certificate) {
+
+
+  let res= JSON.stringify(certificate)
+  			
+  let iexecCallback = ethers.utils.defaultAbiCoder.encode(['string'], [res]);
+  let iexecDeterminism = ethers.utils.keccak256(iexecCallback);
+  fs.writeFile(callbackFilePath, iexecCallback , (err) => {});
+  fs.writeFile(determinismFilePath, iexecDeterminism, (err) => {});
+
+
+  console.log(certificate)  
+
+});
+
```

Comment:
You didn't get the point of what TLS Notary is ... at all

You are retreiving a certificat and pushing in onchain as a stringified json file. A smart contract cannot process that.
The concept of TLS notary is, given an URL, to:
- Retrieve the file at the URL
- While doing so, audit the ssl channel to provide cryptographic proof the file was retreived by this channel
- Format the proof so that an external auditor can check it
- Upload the file (or hash of file) and the proof

Just providing a certificate achieves nothing